src/TimeoutMigrationTool/ASQ/ASQEndpointMigrator.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace Particular.TimeoutMigrationTool.ASQ
+{
+    using System;
+    using Microsoft.Azure.Cosmos.Table;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Cosmos.Table.Queryable;
+
+    public class ASQEndpointMigrator : ITimeoutsTarget.IEndpointTargetBatchMigrator
+    {
+        public ASQEndpointMigrator(CloudTableClient client, string destinationTimeoutTableName, string stagingTableName)
+        {
+            this.client = client;
+            this.destinationTimeoutTableName = destinationTimeoutTableName;
+            this.stagingTableName = stagingTableName;
+        }
+
+        public async ValueTask<int> StageBatch(IReadOnlyList<TimeoutData> timeouts, int batchNumber)
+        {
+            await PurgeStagingTable();
+
+            var stagedTimeouts = timeouts.Select(timeout => StagedDelayedMessageEntity.FromTimeoutData(timeout, batchNumber)).GroupBy(x => x.PartitionKey);
```

Comment:
would something like `timeoutsToBeStaged` be a better name?

Reply:
:+1: